examples/Billing/AddBillingSetup.cs

```diff
@@ -0,0 +1,205 @@
+
+                SetBillingSetupStartDateTime(customerId, googleAdsService, billingSetup);
+
+                // Create the billing setup operation and send it to the BillingSetupService.
+                BillingSetupOperation operation = new BillingSetupOperation()
+                {
+                    Create = billingSetup
+                };
+
+                MutateBillingSetupResponse billingResponse =
+                    billingSetupServiceClient.MutateBillingSetup(customerId.ToString(), operation);
+
+                Console.WriteLine("Added new billing setup with resource name: " +
+                                  $"{billingResponse.Result.ResourceName}");
+            }
+            catch (GoogleAdsException e)
+            {
+                Console.WriteLine("Failure:");
+                Console.WriteLine($"Message: {e.Message}");
+                Console.WriteLine($"Failure: {e.Failure}");
+                Console.WriteLine($"Request ID: {e.RequestId}");
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Creates and returns a new Billing Setup instance with complete payment details. One of
+        /// paymentsAccountId or paymentsProfileId must be provided.
+        /// </summary>
+        /// <param name="customerId">The Google Ads customer ID for which the call is made.</param>
+        /// <param name="paymentsAccountId">Optional payments account ID to attach to the new
+        ///     billing setup. Must be formatted as "1234-5678-9012-3456".</param>
+        /// <param name="paymentsProfileId">Optional payments profile ID to attach to a new payments
+        ///     account and to the new billing setup. Must be formatted as "1234-5678-9012".</param>
+        /// <returns>A new BillingSetup instance with complete payment details.</returns>
+        /// <exception cref="Exception">Generic exception if no payment details have been
+        ///     provided.</exception>
+        private BillingSetup CreateBillingSetup(long customerId, string paymentsAccountId,
+            string paymentsProfileId)
+        {
+            BillingSetup billingSetup = new BillingSetup();
+
+            // Set the appropriate payments account field.
+            if (paymentsAccountId != null)
+            {
+                // If a payments account id has been provided, set PaymentsAccount to its resource
+                // name. You can list available payment accounts via the PaymentAccountService's
+                // ListPaymentsAccounts method.
+                billingSetup.PaymentsAccount =
+                    ResourceNames.PaymentsAccount(customerId, paymentsAccountId);
+            }
+            else
+            {
+                // Otherwise, create a new payments account by setting the PaymentsAccountInfo
+                // field. See https://support.google.com/google-ads/answer/7268503 for information
+                // about payments profiles.
+                if (paymentsProfileId == null)
```

Comment:
nit: it may be subjective but I think that you could use an `else if(paymentsProfileId != null)` and `else ... throw new Exception` to make it more readable.

Reply:
Good suggestion.  Updated.